src/main/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ProfessionalRefreshOrchestrationHelper.java

```diff
@@ -0,0 +1,337 @@
+
+        filteredOrganisationProfiles =
+                extractOrganisationProfiles(filteredOrganisationProfiles,filteredUserAccessTypes);
+
+        usersRoleAssignments = createRoleAssignments(userRefreshQueue, filteredOrganisationProfiles);
+
+        return usersRoleAssignments.stream().toList();
+    }
+
+    private Set<RoleAssignment> createRoleAssignments(UserRefreshQueueEntity userRefreshQueue,
+                                                      Set<OrganisationProfile> organisationProfiles) {
+        Set<RoleAssignment> usersRoleAssignments = new HashSet<>();
+        for (OrganisationProfile organisationProfile:organisationProfiles) {
+            Set<OrganisationProfileJurisdiction> organisationProfileJurisdictions =
+                    organisationProfile.getJurisdictions();
+            for (OrganisationProfileJurisdiction orgProfileJurisdictions : organisationProfileJurisdictions) {
+                String jurisdictionId = orgProfileJurisdictions.getJurisdictionId();
+                for (OrganisationProfileAccessType organisationProfileAccessType:
+                        orgProfileJurisdictions.getAccessTypes()) {
+                    for (AccessTypeRole accessTypeRole: organisationProfileAccessType.getRoles()) {
+                        String organisationalRoleName = accessTypeRole.getOrganisationalRoleName();
+                        RoleAssignment roleAssignment =
+                                createRoleAssignment(organisationalRoleName,userRefreshQueue.getUserId(),jurisdictionId,
+                                        accessTypeRole.getCaseTypeId(),null);
+                        usersRoleAssignments.add(roleAssignment);
+                        if (accessTypeRole.isGroupAccessEnabled()
+                                && StringUtils.isNotBlank(accessTypeRole.getGroupRoleName())
+                                && StringUtils.isNotBlank(accessTypeRole.getCaseGroupIdTemplate())) {
+                            String roleName = accessTypeRole.getGroupRoleName();
+                            String caseAccessGroupId =
+                                    generateCaseAccessGroupId(accessTypeRole.getCaseGroupIdTemplate(),
+                                            userRefreshQueue.getOrganisationId());
+                            RoleAssignment groupRoleAssignment =
+                                    createRoleAssignment(roleName,userRefreshQueue.getUserId(),jurisdictionId,
+                                            accessTypeRole.getCaseTypeId(),caseAccessGroupId);
+                            usersRoleAssignments.add(groupRoleAssignment);
+                        }
+                    }
+                }
+            }
+        }
+        return usersRoleAssignments;
+    }
+
+
+    private Set<OrganisationProfile> extractOrganisationProfiles(Set<OrganisationProfile> organisationProfiles,
+                                                                 List<UserAccessType> userAccessTypes) {
```

Comment:
could this result in some access types that are not required for the user? as you only checking `isAccessMandatory()` || `isAccessDefault()` ...